### ISSUE TITLE
[Proposal] Standard should separate the responsibilities of formatting and linting

### ISSUE THREAD

**Issue by JoshTGreenwood** [testdouble/standard#13]
I want to be able to confidently run `standard . --fix` or even `standard .` and be assured that my project will be conferment and that I will have to do no work. Ideally, this would happen in my editor or a pre-commit hook. It is my belief that if a rule is not auto correctable, it should not be included in the required set of rules. There is still value in some non fixable rules, but they should not be required. 

In my experience, rubocop and like minded tools get run very infrequently and often just on CI. In my opinion, this is the very worst time to ask a user to make changes and invites folks who just want to get their PR merged to make less than ideal choices about how to resolve violations that arise. If we limit standard to only auto-correctable violations, I suspect it will get run by most folks very frequently (ideally via an editor plugin). This ensures compliance at all times and completely removes the 11th hour pressure to make poor code quality choices.

There is prior art here as well. The sort of tool I'm describing is becoming commonplace in other languages with tools like Elm's `elm-format`, Elixir's `mix format`, and Go's `gofmt`.

**Comment by searls** [comment]
tl;dr I am open to it but I don't think it's an easy decision

A few people suggested this last week and I find it compelling… until you see that so many of the most important lint rules (from a code safety perspective) are not autocorrectable. Before solidifying your view on this I'd encourage you to inspect the currently-enabled cops that are not autocorrectable.

My sense:

Pros:
- what you said
- would goose adoption

Cons
- we would be turning off a metric boatload of probably-good rules

Regarding this comment:

>  There is still value in some non fixable rules, but they should not be required.

In case you're implying it, I feel uncomfortable in having two "modes" for standard, one autofixable and one not. It's not very standard if it differs based on how you run it. StandardJS's rules are not all fixable by default for, I think, similar reasons and trade-offs

**Comment by jasonkarns** [comment]
I agree that it would be nice for a ruby formatter. However, I think that's a different project and firmly different goal than that of StandardRB.

StandardJS exists and has many lint rules above and beyond what can be auto formatted by the gofmt-of-JS (prettier). And of standard's rules, only a subset are autofixable. I don't think there's a reason to _not_ auto fix a rule, simply because we can't auto fix them all.

If prettier-rb or rbfmt existed, I think standardrb would still need to exist, and even still autofix whatever it can. (And since one of the primary goals of standardrb is to not be configurable, there really isn't a required/optional ruleset. Standard lints according to its opinion, and should fix as much as it can.)

**Comment by searls** [comment]
After conversations out of band at RubyConf, I think Jason's sense is confirmed (so far). Closing for now
